lib/application.ts

```diff
@@ -1388,22 +1405,39 @@ export class SNApplication {
     }
   }
 
-  public async removePasscode(): Promise<void> {
+  /**
+   * @returns whether the passcode was successfuly removed or not
+   */
+  public async removePasscode(): Promise<boolean> {
+    const passcode = await this.challengeService.promptForPasscode(
+      ChallengeReason.RemovePasscode
+    );
+    if (isNullOrUndefined(passcode)) return false;
+
     const dismissBlockingDialog = await this.alertService.blockingDialog(
       DO_NOT_CLOSE_APPLICATION,
       REMOVING_PASSCODE,
     );
     try {
       await this.removePasscodeWithoutWarning();
+      return true;
     } finally {
       dismissBlockingDialog();
     }
   }
 
+  /**
+   * @returns whether the passcode was successfuly changed or not
+   */
   public async changePasscode(
-    passcode: string,
+    newPasscode: string,
     origination = KeyParamsOrigination.PasscodeChange
-  ): Promise<void> {
+  ): Promise<boolean> {
+    const passcode = await this.challengeService.promptForPasscode(
+      ChallengeReason.ChangePasscode
+    );
+    if (!passcode) return false;
```

Comment:
On line 1415 you use `if (isNullOrUndefined(passcode)) return false;` but here it's a direct check. Not sure if there's a practical difference.

Reply:
There is none but the intent is clearer. I'll amend this

Reply:
There's no difference but it makes the check clearer so I will amend this change